platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java

```diff
@@ -530,10 +534,15 @@ public class SwirldsPlatform implements Platform, Startable {
         // doesn't track ancient events
         validators.add(new AncientValidator(consensusRef::get));
         validators.add(new EventDeduplication(isDuplicateChecks, eventIntakeMetrics));
-        validators.add(StaticValidators.buildParentValidator(initialAddressBook.getSize()));
+        validators.add(StaticValidators.buildParentValidator(currentAddressBook.getSize()));
         validators.add(new TransactionSizeValidator(transactionConfig.maxTransactionBytesPerEvent()));
+        // some events in the PCES might have been created by nodes that are no longer in the current
+        // address book but are in the previous one, so we need both for signature validation
+        final List<AddressBook> validationAddressBooks = Stream.of(currentAddressBook, previousAddressBook)
+                .filter(Objects::nonNull)
+                .toList();
         if (basicConfig.verifyEventSigs()) {
-            validators.add(new SignatureValidator(initialAddressBook));
+            validators.add(new SignatureValidator(validationAddressBooks, CryptoStatic::verifySignature));
```

Comment:
If I understand what this is doing, this introduces a vulnerability. After we replay the preconsensus event stream, this causes us to accept events created by nodes that are no longer in the address book.

Also, this is incompatible with a node's signing key changing, since both keys would take up the same slot in `keyMap`.

I think the proper way to do this is to use the old address book to validate the PCES, then to switch this out for the new address book.

Reply:
- we will keep accepting events from this old node, but its weight will be `0`, it will have to work this way until we have full dynamic AB
- for this reason and some others, we said we would not support changing keys. if you want to change keys, you get a new ID
- we cannot use this AB for just PCES replay, because some nodes might have events that others do not, so they will gossip them